src/lib/containers/IconSvg.tsx

```diff
@@ -1,63 +1,67 @@
 import React, { useEffect } from 'react'
-import { ReactSVG } from 'react-svg'
-import styled, { css } from 'styled-components'
+import {
+  Check,
+  Cached,
+} from '@material-ui/icons'
+import {
+  Data,
+  DataLocked,
+} from '../assets/themed_icons'
 
 export type IconSvgOptions = {
   icon: string
   color: string
-  size: string
+  size?: string
   padding?: 'left' | 'right'
   hoverEffect?: boolean  // not implement currently
 }
 
 export type IconSvgProps = {
-  iconBasePath?: string
   options: IconSvgOptions
 }
 
-const defaultIconPath = '../src/lib/assets/icons/'
-const IconPath = {
-  'data': 'data.svg',
-  'reload': 'reload.svg',
-  'checkMark': 'check-mark.svg',
-  'dataLocked': 'data-locked.svg',
+export type SVGStyleProps = {
+  color?: string
+  fontSize?: string
+  verticalAlign?: string
 }
 
-// Help standardize the sizing of icons
-const IconSize = {
-  'sm': '14px'
-}
+const getIcon = (options:IconSvgOptions) => {
+  const { icon, color } = options
 
-// Help standardize the left or right padding of icons
-const IconPadding = {
-  'sm': '7px'
-}
+  // Styles for svg imported from mui
+  const muiSvgStyle:SVGStyleProps = {
+    color: color,
+    verticalAlign: 'middle',
+  }
+  // Styles for custom svg missing from mui
+  const customSvgStyle:SVGStyleProps = {
+    verticalAlign: 'middle',
+  }
 
-// styled component needs to be outside of the function component and render function
-// to remove multiple render warning
-const StyledSVGIcon = styled(ReactSVG)<IconSvgOptions>`
-  svg {
-    padding: 0;
-    width: auto;
-    ${({size, color, padding}) => {
-      return css`
-        fill: ${color};
-        height: ${IconSize[size]};
-        padding-left: ${padding === 'left' ? IconPadding[size] : 0};
-        padding-right: ${padding === 'right' ? IconPadding[size] : 0};
-        path {
-          fill: ${color};
-        }
-      `
-    }
+  switch (icon) {
```

Comment:
Ok, so this switch maps the icon name to either a custom or material UI icon.  Very nice!